packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -1,27 +1,62 @@
 <script lang="ts" setup>
-import { computed } from "vue";
-
-const props = withDefaults(
-  defineProps<{
-    /** The current input value */
-    modelValue?: string;
-    /** Label to show next to the input */
-    label?: string;
-    required?: boolean;
-  }>(),
-  {
-    modelValue: "",
-    label: "",
-  },
-);
+import {
+  useFormValidationMessage,
+  type SupportedErrorLangs,
+  type FormElementProps,
+} from "@/composables/formValidationMessage";
+import { computed, onMounted, ref, watch } from "vue";
+
+export type TestInputProps = {
+  /** The current input value
+   * TODO: remove the "number" once we separated number inputs from the other types.
+   */
+  modelValue?: string | number;
+  /** Label to show next to the input */
+  label?: string;
+  /** A custom error message that will be shown (only when the input is invalid) */
+  customErrorMessage?: string;
+  /** The selected language key, used for error messages of invalid inputs.
+   * If empty, error messages will default to the browser language. */
+  lang?: SupportedErrorLangs;
+} & FormElementProps;
+
+const props = withDefaults(defineProps<TestInputProps>(), {
+  modelValue: "",
+  label: "",
+  type: "text",
+});
 
 const emit = defineEmits<{
-  /** Emitted when input value changes */
-  "update:modelValue": [value: string];
+  /** Emitted when input value changes
+   * TODO: remove the "number" once we separated number inputs from the other types.
+   */
+  "update:modelValue": [value: string | number];
   /** Emitted on blur when the input value changes */
   change: [value: string];
+  /** Emitted whenever the validity state of the input changes */
+  validityChange: [state: ValidityState];
 }>();
 
+const isTouched = ref(false);
+
+const coreElement = ref<HTMLInputElement | null>(null);
+
+const validityState = ref<ValidityState | undefined>(coreElement.value?.validity);
+
+const errorMessage = computed(() => {
+  /* when the validity state is uninitialized or the form is valid, we don't show an error. */
+  if (!validityState.value || validityState.value.valid) return "";
+  /* a custom error message always is considered first */
+  if (props.customErrorMessage) return props.customErrorMessage;
+
+  const element = coreElement.value;
+  /* when a language key is provided, we offer our own translations of the error messages 
+  to match the rest of the user's application */
+  if (props.lang && element) return useFormValidationMessage(props.lang, element.validity, props);
```

Comment:
The way the `useFormValidationMessage` is used is rather a utility than a composable since it has no state

Reply:
what would you recommend
1. put the result of useFormValidationMessage in a separate computed which is referenced here
2. change it to getFormValidationMessage 
-> question on 2): can MaybeRefOrGetter also be used on a util?

Reply:
I think its fine to leave it like so for now until we discuss #71 tomorrow